src/relayclient/RelayClient.ts

```diff
@@ -108,16 +139,18 @@ export class RelayClient {
   }
 
   async _init (): Promise<void> {
+    if (this.initialized) { return }
+    this.emit(new GsnInitEvent())
     await this.contractInteractor.init()
     this.initialized = true
   }
 
   async relayTransaction (gsnTransactionDetails: GsnTransactionDetails): Promise<RelayingResult> {
-    if (!this.initialized) {
-      await this._init()
-    }
+    await this._init()
     // TODO: should have a better strategy to decide how often to refresh known relays
+    this.emit(new GsnRefreshRelaysEvent())
     await this.knownRelaysManager.refresh()
+    this.emit(new GsnRefreshedRelaysEvent((await this.knownRelaysManager.getRelaysSortedForTransaction({} as any)).length))
```

Comment:
WTF?

Reply:
I understand the API did not suit your use-case so you abused the API with a cast of empty object to any.
I assume this is a good place to make a change in the API.

Reply:
tried to limit the pr to events only. but yes, API change is needed here, to expose the list of relays.

Reply:
Note that `knownRelaysManager.knownRelays` seems to be accessible. Is it not what you are looking for? I am surprised this does not crash. It will crash one day.